cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/DefaultPreviewRunnerManager.java

```diff
@@ -134,14 +134,15 @@ protected void startUp() throws Exception {
 
   @Override
   protected void shutDown() throws Exception {
+    // Should stop the polling service, hence individual preview runs, before stopping the top level preview runner.
```

Comment:
Correct. I noticed this in the testing as well. I think we should change the start order as well. Start the runner before starting pollers.

Reply:
Moved